streamingvisitors/src/vespa/searchvisitor/searchvisitor.cpp

```diff
@@ -739,12 +740,12 @@ void SearchVisitor::setupAttributeVector(const FieldPath &fieldPath) {
     enum FieldDataType { OTHER = 0, ARRAY, WSET };
     FieldDataType typeSeen = OTHER;
     for (const auto & entry : fieldPath) {
-        int dataTypeId(entry->getDataType().getClass().id());
-        if (dataTypeId == document::ArrayDataType::classId) {
+        const document::DataType & dt = entry->getDataType();
+        if (dt.isArray()) {
             typeSeen = ARRAY;
-        } else if (dataTypeId == document::MapDataType::classId) {
+        } else if (dt.cast_map() != nullptr) {
```

Comment:
Should there be a distinct `isMap()` for symmetry?

Reply:
Fixed